frontend/src/pages/Dashboard/ResearcherDashboard.tsx

```diff
@@ -0,0 +1,241 @@
+
+  @media (max-width: 900px) {
+    min-width: 100%;
+    grid-template-columns: 1fr 1fr;
+    column-gap: 24px;
+    row-gap: 24px;
+  }
+`;
+
+const ResearcherTopBar = styled.div`
+  flex: 7;
+  display: flex;
+  justify-content: space-between;
+  margin-bottom: 24px;
+
+  @media (max-width: 375px) {
+    flex-wrap: wrap;
+  }
+`;
+
+const StudentCardContainer = styled.div`
+  background: ${SKY};
+  border-radius: 12px;
+  padding: 32px 10px;
+  text-align: center;
+`;
+
+const TitleText = styled.p`
+  font-family: "Rubik";
+  font-size: 26px;
+  font-weight: 700;
+  text-transform: lowercase;
+
+  @media (max-width: 900px) {
+    font-size: 22px;
+  }
+`;
+
+const Avatar = styled.img`
+  border-radius: 50%;
+
+  @media (max-width: 900px) {
+    width: 62px;
+    height: 62px;
+  }
+`;
+
+const StudentCardLabel = styled.p`
+  font-family: Rubik;
+  font-size: 18px;
+  font-style: normal;
+  font-weight: 700;
+  line-height: 28px;
+  letter-spacing: 0px;
+  text-align: center;
+  margin-top: 16px;
+  @media (max-width: 900px) {
+    font-size: 16px;
+    line-height: 25px;
+    margin-top: 10px;
+  }
+`;
+
+const StudentFilter = styled(Dropdown)`
+  background: ${CLOUD};
+  padding: 10px;
+  border-radius: 12px;
+`;
+
+const DropdownLabel = styled.div`
+  font-size: 18px;
+  font-weight: 700;
+  height: 50px;
+  width: 260px;
+  display: flex;
+  padding: 10px 24px;
+  justify-content: space-between;
+  align-items: baseline;
+  cursor: pointer;
+
+  @media (max-width: 900px) {
+    font-size: 16px;
+    width: 220px;
+    padding: 10px 18px;
+  }
+`;
+
+const DropdownCaret = styled.img`
+  margin: 5px 0px 5px 5px;
+  padding-bottom: 2px;
+  width: 15px;
+  height: 10px;
+
+  @media (max-width: 900px) {
+    width: 12px;
+    height: 6px;
+  }
+`;
+
+const DropdownMenu = styled(Menu)`
+  background: ${CLOUD};
+  border-radius: 12px;
+  li {
+    font-size: 18px;
+  }
+
+  @media (max-width: 900px) {
+    li {
+      font-size: 16px;
+    }
+  }
+`;
+
+interface StudentCardParams {
+  student: User;
+}
+
+const StudentCard: FunctionComponent<StudentCardParams> = ({
+  student,
+}) => {
+  return (
+    <StudentCardContainer>
+      <Avatar
+        src={
+          student.profileIcon
+            ? student.profileIcon
+            : "https://firebasestorage.googleapis.com/v0/b/vocab-buddy-53eca.appspot.com/o/icons%2Frocketcircle.svg?alt=media&token=2378a891-001b-4a7f-9522-3dffee8d202d"
+        }
+      />
+      <StudentCardLabel>{student.name}</StudentCardLabel>
+    </StudentCardContainer>
+  );
+};
+
+const sortByName = (studentA: User, studentB: User): number => {
+  const aName = studentA.name.toLowerCase();
+  const bName = studentB.name.toLowerCase();
+  if (aName > bName) {
+    return 1;
+  } else if (bName > aName) {
+    return -1;
+  }
+  return 0;
+};
+
+const sortByAge = (studentA: User, studentB: User): number => {
+  return studentA.age - studentB.age;
+};
+
+const sortBySession = (studentA: User, studentB: User): number => {
```

Comment:
I'm seeing some weird behavior when resizing the screen while sorting by session.

 I don't know why this is happening exactly, but an idea for fix:
- a sort by name as a secondary sort in the case that the students are at the same session



https://user-images.githubusercontent.com/17476012/112082721-378a6180-8b5c-11eb-807d-9677e49d7369.mp4

Reply:
I do see that, great catch!

Reply:
good catch! we didn't check if both session ids were undefined. this fixed it for me on chrome, let me know if it works for you!